### PR TITLE
[fix](cast) return error when numerical cast out-of-range

### PR DESCRIPTION
**Pull request by kaijchen**
## Proposed changes

Return an error when numerical cast result is out-of-range.

Before:
image

After:
image

Note: [MySQL no restrictive modes](https://dev.mysql.com/doc/refman/8.0/en/out-of-range-and-overflow.html) is not supported.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by kaijchen** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by dataroaring** [comment]
handle streamload and add outof range case for all load.

File: be/src/vec/functions/function_cast.h [hunk 1]
```diff
@@ -432,6 +432,16 @@ struct ConvertImpl {
                     block.get_by_position(result).column =
                             ColumnNullable::create(std::move(col_to), std::move(col_null_map_to));
                     return Status::OK();
+                } else if constexpr (IsDataTypeNumber<FromDataType> &&
+                                     IsDataTypeNumber<ToDataType>) {
+                    for (size_t i = 0; i < size; ++i) {
+                        if (vec_from[i] < min_result || vec_from[i] > max_result) {
+                            return Status::InternalError(
+                                    "Out of range value for column '{}' at row {}",
```
Comment (yiguolei) [inline be/src/vec/functions/function_cast.h:440]: 不要这么写error message， 实际这里的行号是错误的。
写column name 也不一定对，这里可能没有name。
这里只写错误的值吧。

Comment (kaijchen) [reply be/src/vec/functions/function_cast.h]: 改成了 Postgres style 的 `Error: SMALLINT out of range`

**Pull request closed without merging**